packages/twenty-server/src/engine/core-modules/billing/services/billing-portal.workspace-service.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+
+import { BillingSubscription } from 'src/engine/core-modules/billing/entities/billing-subscription.entity';
+import { BillingSubscriptionService } from 'src/engine/core-modules/billing/services/billing-subscription.service';
+import { StripeService } from 'src/engine/core-modules/billing/stripe/stripe.service';
+import { UserWorkspaceService } from 'src/engine/core-modules/user-workspace/user-workspace.service';
+import { User } from 'src/engine/core-modules/user/user.entity';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { EnvironmentService } from 'src/engine/integrations/environment/environment.service';
+import { assert } from 'src/utils/assert';
+
+export enum WebhookEvent {
+  CUSTOMER_SUBSCRIPTION_CREATED = 'customer.subscription.created',
+  CUSTOMER_SUBSCRIPTION_UPDATED = 'customer.subscription.updated',
+  CUSTOMER_SUBSCRIPTION_DELETED = 'customer.subscription.deleted',
+  SETUP_INTENT_SUCCEEDED = 'setup_intent.succeeded',
+}
+
+@Injectable()
+export class BillingPortalWorkspaceService {
+  protected readonly logger = new Logger(BillingPortalWorkspaceService.name);
+  constructor(
+    private readonly stripeService: StripeService,
+    private readonly userWorkspaceService: UserWorkspaceService,
+    private readonly environmentService: EnvironmentService,
+    @InjectRepository(BillingSubscription, 'core')
+    private readonly billingSubscriptionRepository: Repository<BillingSubscription>,
+    private readonly billingSubscriptionService: BillingSubscriptionService,
+  ) {}
+
+  async computeCheckoutSessionURL(
+    user: User,
+    workspace: Workspace,
+    priceId: string,
+    successUrlPath?: string,
+  ): Promise<string> {
+    const frontBaseUrl = this.environmentService.get('FRONT_BASE_URL');
+    const successUrl = successUrlPath
+      ? frontBaseUrl + successUrlPath
+      : frontBaseUrl;
+
+    const quantity =
+      (await this.userWorkspaceService.getUserCount(workspace.id)) || 1;
+
+    const stripeCustomerId = (
+      await this.billingSubscriptionRepository.findOneBy({
+        workspaceId: user.defaultWorkspaceId,
```

Comment:
@charlesBochet again I don't know this part very well but here we have both workspace.id and user.defaultWorkspaceId. Could this lead to issue? Shouldn't we use always the same value?

Reply:
yes this is not great. It's working right now because we change the defaultWorkspaceId on login / switch
We will need to add a workspaceId param later